charts/terra-argocd-app/templates/app.yaml

```diff
@@ -1,39 +1,45 @@
-{{- with .Values -}}
+{{- if .Values.releaseType | eq "app" }}
+# Generates an ArgoCD application for a Terra application release
+{{- $app := required "releaseName is required" .Values.releaseName }}
```

Comment:
Is there a reason we can't use `Release.Name` [link](https://helm.sh/docs/chart_template_guide/builtin_objects/)?